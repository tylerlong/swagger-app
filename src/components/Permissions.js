import React from 'react'
import R from 'ramda'
import { Form, Button, Collapse } from 'antd'
import { connect } from 'react-redux'

import PermissionFormItem from './PermissionFormItem'
import { addPermission, setProp } from '../actions'

class Permissions extends React.Component {
  render () {
    return (
      <div>
        <h2>Permissions</h2>
        <Form>
          <Collapse accordion activeKey={`${this.props.metadata.activePermissionKey}`} onChange={key => { this.props.setProp(['metadata', 'activePermissionKey'], key) }}>
            {this.props.permissions.map((permission, index) => {
              return (
                <Collapse.Panel header={permission.name} key={`${this.props.permissions.length}-${index}`}>
                  <PermissionFormItem index={index} />
                </Collapse.Panel>
              )
            })}
          </Collapse>
        </Form>
        <div className='primary-button'>
          <Button type='primary' size='large' onClick={this.props.addPermission}>Add permission</Button>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { addPermission, setProp })(Permissions)
