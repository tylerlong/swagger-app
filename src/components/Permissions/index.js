import React from 'react'
import R from 'ramda'
import { Button, Collapse, Icon } from 'antd'
import { connect } from 'react-redux'

import FormItem from './FormItem'
import { addPermission } from '../../actions'

class Permissions extends React.Component {
  render () {
    console.log(`render Permissions`)
    const { permissions, addPermission } = this.props
    return (
      <div>
        <h2>Permissions</h2>
        <Collapse accordion>
          {R.sort(R.comparator((a, b) => R.toLower(a.name) < R.toLower(b.name)), permissions).map(permission => {
            return (
              <Collapse.Panel header={permission.name} key={`${permission.createdAt}`}>
                <FormItem index={R.findIndex(R.propEq('createdAt', permission.createdAt), permissions)} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <Button type='primary' size='large' onClick={addPermission}><Icon type='plus' />Add permission</Button>
      </div>
    )
  }
}

export default connect(R.pick(['permissions']), { addPermission })(Permissions)
