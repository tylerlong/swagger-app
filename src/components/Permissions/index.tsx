import React from 'react'
import R from 'ramda'
import { Button, Collapse, Icon } from 'antd'
import { connect } from 'react-redux'

import FormItem from './FormItem'
import { addPermission } from '../../actions'
import { orderBy } from '../../utils'

class Permissions extends React.Component<any> {
  render() {
    console.log(`render Permissions`)
    const { permissions, addPermission } = this.props
    return (
      <div>
        <h2>Permissions</h2>
        <Collapse accordion>
          {orderBy(R.prop('name'), permissions).map(permission => {
            return (
              <Collapse.Panel header={permission.name} key={`${permission.createdAt}`}>
                <FormItem index={R.findIndex(R.propEq('createdAt', permission.createdAt), permissions)} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <Button type='primary' size='large' onClick={addPermission}><Icon type='plus' />Add permission</Button>
      </div>
    )
  }
}

export default connect(R.pick(['permissions']), { addPermission })(Permissions)
