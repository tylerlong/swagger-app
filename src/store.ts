import { createStore, applyMiddleware } from 'redux'
import { createLogicMiddleware } from 'redux-logic'

import reducer from './reducers'
import logics from './logics'

const logicMiddleware = createLogicMiddleware(logics)
const store = createStore(
  reducer,
  applyMiddleware(logicMiddleware)
)

export default store
