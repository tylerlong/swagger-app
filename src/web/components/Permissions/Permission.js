import React from 'react'

import { PermissionTextField, DeletePermissionButton } from '../../containers/Permissions/Permission'
import { pathType } from '../../utils'

class Permission extends React.Component {
  render () {
    console.log(`render Permission`)
    const { path } = this.props
    return (
      <div>
        <DeletePermissionButton path={path} />
        <PermissionTextField path={path} name='name' />
        <PermissionTextField path={path} name='description' />
      </div>
    )
  }
}

Permission.propTypes = {
  path: pathType
}

export default Permission
