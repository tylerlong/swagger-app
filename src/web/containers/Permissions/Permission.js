import R from 'ramda'
import { connect } from 'react-redux'

import { setProp, deleteFromArray } from '../../actions'
import { TextField, DeleteButton } from '../../components/Common'

export const PermissionTextField = connect(
  (state, { path, name }) => ({ value: R.path(path.concat(name), state) }),
  (dispatch, { path, name }) => ({
    update: value => dispatch(setProp(path.concat(name), value))
  }))(TextField)

export const DeletePermissionButton = connect(
  (state, { path }) => ({
    recordName: R.path(path.concat('name'), state)
  }),
  (dispatch, { path }) => ({
    deleteRecord: () => dispatch(deleteFromArray(path))
  }))(DeleteButton)
