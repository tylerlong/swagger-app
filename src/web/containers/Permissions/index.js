import React from 'react'
import R from 'ramda'
import { Button, Collapse, Icon } from 'antd'
import { connect } from 'react-redux'

import Permission from './Permission'
import { addPermission } from '../../actions'
import { orderBy } from '../../utils'

class Permissions extends React.Component {
  render () {
    console.log(`render Permissions`)
    const { permissions, addPermission } = this.props
    return (
      <div>
        <h2>Permissions</h2>
        <Collapse accordion>
          {orderBy(R.prop('name'), permissions).map(permission => {
            return (
              <Collapse.Panel header={permission.name} key={`${permission.createdAt}`}>
                <Permission index={R.findIndex(R.propEq('createdAt', permission.createdAt), permissions)} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <Button type='primary' size='large' onClick={addPermission}><Icon type='plus' />Add permission</Button>
      </div>
    )
  }
}

export default connect(R.pick(['permissions']), { addPermission })(Permissions)
