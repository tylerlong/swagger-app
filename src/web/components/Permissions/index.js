import React from 'react'
import { Collapse } from 'antd'
import PropTypes from 'prop-types'

import { AddPermissionButton } from '../../containers/Permissions'
import Permission from './Permission'
import { objType } from '../../utils'

class Permissions extends React.Component {
  render () {
    console.log(`render Permissions`)
    const { permissions } = this.props
    return (
      <div>
        <h2>Permissions</h2>
        <Collapse accordion>
          {permissions.map(({ path, name, createdAt }) => {
            return (
              <Collapse.Panel header={name} key={createdAt}>
                <Permission path={path} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <AddPermissionButton />
      </div>
    )
  }
}

Permissions.propTypes = {
  permissions: PropTypes.arrayOf(objType)
}

export default Permissions
