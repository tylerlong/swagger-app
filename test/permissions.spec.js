/* eslint-env jest */
import R from 'ramda'
import { Popconfirm } from 'antd'

import Permissions from '../src/web/components/Permissions'
import Permission from '../src/web/components/Permissions/Permission'
import store from './store'
import { getWrapper } from './shared'
import state from '../dist/state.json'

let wrapper = null
beforeEach(() => {
  wrapper = getWrapper(Permissions, R.pick(['permissions'], state))
})
const count = state.permissions.length
const getCount = () => store.getState().permissions.length
const getModel = (index) => store.getState().permissions[index]

describe('test permissions', () => {
  test('permissions list', () => {
    expect(wrapper.find('.ant-collapse-item').length).toEqual(count)
  })

  test('add permission', () => {
    wrapper.find('button.ant-btn-primary').simulate('click')
    expect(getCount()).toEqual(count + 1)
  })

  test('delete permission', () => {
    wrapper.find('div.ant-collapse-header').first().simulate('click')
    wrapper.find(Popconfirm).props().onConfirm()
    expect(getCount()).toEqual(count - 1)
  })

  test('update permission fields', () => {
    wrapper.find('div.ant-collapse-header').first().simulate('click')
    const index = wrapper.find(Permission).first().props().index

    // name
    let input = wrapper.find('input').first()
    input.simulate('change', { target: { value: 'Hello' } })
    expect(getModel(index).name).toEqual('Hello')
    input.simulate('change', { target: { value: 'World' } })
    expect(getModel(index).name).toEqual('World')

    // description
    input = wrapper.find('input').at(1)
    input.simulate('change', { target: { value: 'Hello' } })
    expect(getModel(index).description).toEqual('Hello')
    input.simulate('change', { target: { value: 'World' } })
    expect(getModel(index).description).toEqual('World')
  })
})
