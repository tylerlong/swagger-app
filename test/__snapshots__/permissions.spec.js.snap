// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test permissions list 1`] = `
<Provider
  store={
    Object {
      "actions": Array [],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Permissions)>
    <Permissions
      addPermission={[Function]}
      metadata={
        Object {
          "alerts": Array [
            Object {
              "message": "Data loaded",
              "type": "success",
            },
          ],
        }
      }
      permissions={
        Array [
          Object {
            "createdAt": 1498875025725,
            "description": "Viewing and updating user account info (including name, business name, address and phone number/account number)",
            "name": "EditAccounts",
          },
          Object {
            "createdAt": 1498875008388,
            "description": "Managing accounts: creating new accounts, viewing and updating account information, deleting existing accounts",
            "name": "Accounts",
          },
          Object {
            "createdAt": 1498875020815,
            "description": "Creating, viewing, editing and deleting user personal contacts",
            "name": "Contacts",
          },
        ]
      }
      setProp={[Function]}
    >
      <div>
        <h2>
          Permissions
        </h2>
        <Collapse
          accordion={true}
          bordered={true}
          openAnimation={
            Object {
              "appear": [Function],
              "enter": [Function],
              "leave": [Function],
            }
          }
          prefixCls="ant-collapse"
        >
          <Collapse
            accordion={true}
            bordered={true}
            className=""
            destroyInactivePanel={false}
            onChange={[Function]}
            openAnimation={
              Object {
                "appear": [Function],
                "enter": [Function],
                "leave": [Function],
              }
            }
            prefixCls="ant-collapse"
          >
            <div
              className="ant-collapse"
            >
              <CollapsePanel
                destroyInactivePanel={false}
                header="Accounts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    Accounts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="Contacts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    Contacts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="EditAccounts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    EditAccounts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
            </div>
          </Collapse>
        </Collapse>
        <Button
          clicked={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          size="large"
          type="primary"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg"
            onClick={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <Icon
              type="plus"
            >
              <i
                className="anticon anticon-plus"
              />
            </Icon>
            <span>
              Add permission
            </span>
          </button>
        </Button>
      </div>
    </Permissions>
  </Connect(Permissions)>
</Provider>
`;

exports[`test permissions list 2`] = `
<Provider
  store={
    Object {
      "actions": Array [
        Object {
          "createdAt": 1482363367071,
          "description": "Description",
          "name": "Name",
          "type": "ADD_PERMISSION",
        },
      ],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Permissions)>
    <Permissions
      addPermission={[Function]}
      metadata={
        Object {
          "alerts": Array [
            Object {
              "message": "Data loaded",
              "type": "success",
            },
            Object {
              "message": "Permission added",
              "type": "success",
            },
          ],
        }
      }
      permissions={
        Array [
          Object {
            "createdAt": 1498875025725,
            "description": "Viewing and updating user account info (including name, business name, address and phone number/account number)",
            "name": "EditAccounts",
          },
          Object {
            "createdAt": 1498875008388,
            "description": "Managing accounts: creating new accounts, viewing and updating account information, deleting existing accounts",
            "name": "Accounts",
          },
          Object {
            "createdAt": 1498875020815,
            "description": "Creating, viewing, editing and deleting user personal contacts",
            "name": "Contacts",
          },
          Object {
            "createdAt": 1482363367071,
            "description": "Description",
            "name": "Name",
          },
        ]
      }
      setProp={[Function]}
    >
      <div>
        <h2>
          Permissions
        </h2>
        <Collapse
          accordion={true}
          bordered={true}
          openAnimation={
            Object {
              "appear": [Function],
              "enter": [Function],
              "leave": [Function],
            }
          }
          prefixCls="ant-collapse"
        >
          <Collapse
            accordion={true}
            bordered={true}
            className=""
            destroyInactivePanel={false}
            onChange={[Function]}
            openAnimation={
              Object {
                "appear": [Function],
                "enter": [Function],
                "leave": [Function],
              }
            }
            prefixCls="ant-collapse"
          >
            <div
              className="ant-collapse"
            >
              <CollapsePanel
                destroyInactivePanel={false}
                header="Accounts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    Accounts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="Contacts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    Contacts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="EditAccounts"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    EditAccounts
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="Name"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    Name
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
            </div>
          </Collapse>
        </Collapse>
        <Button
          clicked={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          size="large"
          type="primary"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg ant-btn-clicked"
            onClick={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <Icon
              type="plus"
            >
              <i
                className="anticon anticon-plus"
              />
            </Icon>
            <span>
              Add permission
            </span>
          </button>
        </Button>
      </div>
    </Permissions>
  </Connect(Permissions)>
</Provider>
`;
